b2sdk/sync/folder.py

```diff
@@ -294,6 +308,20 @@ def all_files(self, reporter, policies_manager=DEFAULT_SCAN_MANAGER):
             if policies_manager.should_exclude_file(file_name):
                 continue
 
+            # Do not allow relative paths in file names
+            if (file_name.startswith('..' + os.path.sep) or
+                os.path.sep + '..' + os.path.sep in file_name or
+                file_name.endswith(os.path.sep + '..') or file_name == '..'
+            ):
+                raise Exception(
+                    "sync does not support file names that include relative paths: %s" % file_name
+                )
+            # Do not allow absolute paths in file names
+            if file_name.startswith(os.path.sep) or re.match("[A-Za-z]:" + os.path.sep, file_name):
```

Comment:
There will always be issues stemming from the differences in the namespaces between the  b2 namespace, Windows filesystem namespace, and linux or OSX's filesystem namespace.  Since b2's namespace is, in many ways, more permissive than either Windows, OSX, or LInux, potential issues can arise.

It is up to the users of the b2 sync function to ensure they do not violate the namespace limitations for the the platform namespaces they intend to use.  It's easy to upload a file with a colon (':') in it and break syncing with Windows.  It's even *easier* to go from a case-sensitive namespace to a non-case-sensitive namespace and cause problems.

In this case, we are not trying to solve/avoid/limit namespace issues, we simply wish to prevent the sync function from writing files outside the designated local destination directory.

Reply:
ok, fair point, but then `c:/abc` is not an absolute path on linux and this line is refusing to sync it. I think we need to check the platform and base the allowed paths on the result of that check.

Reply:
Agreed.  The driver-letter matching is now only applied on Windows.